File: math/gcd/gcd.py
"""Math: Greatest Common Divisor (Euclidean algorithm)"""


def gcd(a: int, b: int) -> int:
    while b != 0:
        a, b = b, a % b
    return a


if __name__ == "__main__":
    a, b = 48, 18
    print(f"gcd({a}, {b}) = {gcd(a, b)}")

File: math/gcd/gcd.ts
// Math: Greatest Common Divisor (Euclidean algorithm)

function gcd(a: number, b: number): number {
  while (b !== 0) {
    [a, b] = [b, a % b];
  }
  return a;
}

function main(): void {
  const a = 48;
  const b = 18;
  console.log(`gcd(${a}, ${b}) = ${gcd(a, b)}`);
}

main();
